hlink/linking/core/transforms.py

```diff
@@ -114,9 +123,9 @@ def parse_feature_selections(df_selected, feature_selection, is_a):
             return df_selected
 
         elif transform == "array":
-            col1, col2 = feature_selection["input_columns"]
+            input_cols = feature_selection["input_columns"]
             output_col = feature_selection["output_column"]
-            df_selected = df_selected.withColumn(output_col, array(col1, col2))
+            df_selected = df_selected.withColumn(output_col, array(input_cols))
```

Comment:
I don't really know how the pyspark.sql.functions.array() works when a list is passed, though your tests seem to indicate it's fine, but [this](https://www.everythingspark.com/pyspark/working-with-arrays-in-pyspark/) makes it look like you can just pass the list itself as an "array literal". Or does passing it this way allow for the "input_columns" to be either a single string or a list of strings?

Reply:
I didn't think to look too closely at that either since the tests passed.

Pyspark automatically flattens the argument when you pass a single list to `array()`: https://api-docs.databricks.com/python/pyspark/latest/pyspark.sql/api/pyspark.sql.functions.array.html. So `array(input_cols)` and `array(*input_cols)` are equivalent here. I guess that passing it this way allows the `input_columns` in the config to be either a string or list! But that was not the intention of doing it this way.

I think the "array literal" syntax might be out of date, because I get a type error from it.

```
from hlink.spark.factory import SparkFactory

factory = SparkFactory()
spark = factory.create()
df = spark.createDataFrame([[1, 2, 3], [2, 3, 4]], schema=["A", "B", "C"])
df2 = df.withColumn("D", ["A", "B"])
```

gives me

```
pyspark.errors.exceptions.base.PySparkTypeError: [NOT_COLUMN] Argument `col` should be a Column, got list.
```